docs/design/provisioning.md

```diff
@@ -0,0 +1,322 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/design/provisioning.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Managing Persistent Storage
+
+This document proposes a model for the configuration and management of persistent storage in Kubernetes.  Familiarity [Persistent Volumes](../user-guide/persistent-volumes/) is assumed.
+
+## tl;dr
+
+New config kinds:
+
+A `PersistentStorageConfig` represents a cluster's desired behavior when managing persistent volumes and claims.  `PersistentStorageConfig` is comprised of Binders (matching claims to volumes), Provisioners (creating resources on the fly), and Recyclers (reclaiming used resources).
+
+A `BinderConfig` is a pair of selectors that must match a volume and claim in order to make a bind.  Admins use this to earmark volumes for specific purposes.
+
+A `ProvisionerConfig` is a selector for persistent volume claims whose matching volumes will be provisioned on demand.  The config contains a template of the persistent volume to create, the security context under which to create it, and the name of the plugin that performs the operation.
+
+A `RecyclerConfig` is a selector for persistent volumes to recycle, a security context under which to recycle it, and the name of the plugin that performs the operation.
+
+New plugins:
+
+`PersistentVolumeProvisioner` is a plugin that can create real storage with a storage provider (such as EBS or GCE).  This plugin replaces `volume.Creater`.
+
+`PersistentVolumeRecycler` is a plugin that reclaims discarded resources.
+
+### Goals
+
+* Allow admins to earmark volumes to claims with matching label selectors
+* Allow users to request volumes via selector [#14908](https://github.com/kubernetes/kubernetes/issues/14908)
+* Allow admins to configure many provisioners using persistent volume templates
+* Allow admins to configure many recyclers of persistent volumes
+* Decouple provisioning code from volume plugin code
+
+
+### PersistentStorageConfig
+
+Administrators map binders, provisioners, and recyclers to specific types of persistent claims and volumes by creating a PersistentStorageConfig file.  The config file is set as a CLI flag.
+
+Binders contains two selectors against which they match claims for binding.  One selector matches claims and the other selector matches volumes.  Both must be true in order to bind, in addition to other matching semantics (e.g, capacity). Claims can remain unbound indefinitely.
```

Comment:
Will a Binder look for a Provisioner to kick if no matching claims are available?  Also, will a binder update the status of a PVC to indicate that it looked for a match, but couldn't find one?  That could be used to indirectly kick the Provisioner and it would easily allow for adding a provisioner after the fact.